src/csharp/Microsoft.Spark.Worker.UnitTest/CommandExecutorTests.cs

```diff
@@ -25,6 +26,19 @@ namespace Microsoft.Spark.Worker.UnitTest
 {
     public class CommandExecutorTests
     {
+        private readonly Version _version;
+        private readonly IpcOptions _ipcOptions;
+
+        public CommandExecutorTests()
+        {
+            // Spark 3.x and tests use ArrowStreamReader 0.15.1+
```

Comment:
Shouldn't we also test against Spark 2?

Reply:
I guess the question is, how do we test against spark 2?  Spark 2 uses the legacy Arrow spec, however in these tests we're using Arrow 0.15.1, and therefore ArrowStreamReader is expecting the new Arrow spec.

Reply:
set `WriteLegacyIpcFormat` to true when you test against Spark 2?

Reply:
Arrow 0.15.1 is working against Spark 2, no?

Reply:
I just wasn't sure if `ArrowStreamReader` was capable of reading both legacy and non legacy specs.

Reply:
Ok, got it.   Thought we were only using ArrowStreamReader in the tests, but I guess we use it in the SqlCommandExecutor as well.  So it must be smart enough to read both.